web/src/components/customer/listing-details/contexts/CommitContext.tsx

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, {useContext, useState, useEffect} from 'react';
+
+import {getCommits, addCommit, deleteCommit} from 'api';
+import {useCustomerContext} from 'components/customer/contexts/CustomerContext';
+import {useCommitFeedbackPromptContext} from 'components/customer/listing-details/contexts/CommitFeedbackPromptContext';
+import {CommitStatus} from 'interfaces';
+
+type ContextType =
+  | {
+      commitStatus: CommitStatus | undefined;
+      onCommit: () => Promise<void>;
+      onUncommit: () => Promise<void>;
+    }
+  | undefined;
+
+const CommitContext = React.createContext<ContextType>(undefined);
+
+/**
+ * useContext hook that ensures it is used within a CommitContext.
+ */
+const useCommitContext = () => {
+  const context = useContext(CommitContext);
+  if (context === undefined) {
+    throw new Error('useCommitContext must be used within a CommitContext');
+  }
+  return context;
+};
+
+interface CommitContextProps {
+  listingId: number;
+}
+
+/**
+ * CommitContextProvider provider with stateful commit details.
+ */
+const CommitContextProvider: React.FC<CommitContextProps> = ({
+  children,
+  listingId,
+}) => {
+  const {
+    customer,
+    idToken,
+    getCustomerWithLogin,
+    refetchCustomer,
+  } = useCustomerContext();
+
+  const {onOpen: onOpenPrompt} = useCommitFeedbackPromptContext();
+
+  const [commitStatus, setCommitStatus] = useState<CommitStatus>();
+  const [commitId, setCommitId] = useState<number | undefined>();
+
+  useEffect(() => {
+    if (customer === undefined) {
+      setCommitStatus(undefined);
+      return;
+    }
+
+    const fetchCommit = async () => {
+      const [commit] = await getCommits({
+        listingId,
+        customerId: customer.id,
+      });
+      if (commit !== undefined) {
```

Comment:
Is it possible for this to be undefined?

Reply:
@chshda Yep, if there are no commits returned from `getCommits` then `commit` will be undefined!
When I do 
```
const [commit] = await getCommits(...);
```
This means that I am doing something equivalent to 
```
const commit = await getCommits(...)[0];
```
So if `getCommits` returns an empty array, `commit` will be undefined